Consistent layout (#247)

Simon-Deuring opened the pull request:
### New
- Consistent layout for all views
### Changes
- All HTML tags use Bootstrap if possible
- Input validation for all fields
- No inline CSS
- Describing comments for all views, components and methods

Commit:
Consistent layout for all views. Add comments to HTML code

Commit:
Add Bootstrap elements and comments to the CMS

Commit:
Add Bootstrap elements and comments to all editors

Commit:
Add last missing Bootstrap elements and comments

Commit:
Remove last inconsistencies

Commit:
Remove all inline CSS

Commit:
Add title validation for all editors

Commit:
Small modification in the presentation of files

DevPhilB commented on vue/src/components/BoardSidebar.vue:
```diff
@@ -1,3 +1,4 @@
+<!-- This sidebar displays all notice board accessable for the user -->
```
accessable != accessible

DevPhilB commented on vue/src/components/BoardSidebar.vue:
```diff
@@ -93,6 +94,7 @@ export default {
     this.loading = false
   },
   methods: {
+    // Used to load the all boards accessable for the user
```
Cf. first comment (change it to accessible)

DevPhilB commented on vue/src/components/NoteFooter.vue:
```diff
@@ -1,3 +1,4 @@
+<!-- This footer is displayed beneath every note. The deletion button only is shown if the user is authorized to delete it -->
```
"This footer is displayed under the content of every note. The deletion button is only shown if the user is authorized to delete it" or just "This footer is part of every note. The deletion button is only shown if the user is authorized to delete it"

DevPhilB commented on vue/src/components/cms/PermissionPanel.vue:
```diff
@@ -1,51 +1,69 @@
+<!-- This panel shows a list of user rights. They can be granted and deprived separately -->
```
"This panel shows a list of user rights. They can be given and revoked separately"

DevPhilB commented on vue/src/components/cms/UserList.vue:
```diff
@@ -1,3 +1,5 @@
+<!-- This panel shows a list of all registered users. For each user the profil page can be visited and edited.
+     It also allows to promote, demote and delte users -->
```
"This panel shows a list of all registered users. Each user profil can be visited directly in view- or edit-mode.
"It also allows to promote, demote and delete users"

DevPhilB commented on vue/src/components/cms/UserList.vue:
```diff
@@ -322,33 +328,38 @@ export default {
-      this.deleteUser(this.selectedId)
-    },
-    clickProfile () {
+    // This method opens the profile page in editing mode
```
This method opens the profile page in edit mode

DevPhilB commented on vue/src/components/editors/EditorIcon.vue:
```diff
@@ -1,3 +1,4 @@
+<!-- This class is used to draw the icons visible beneath the text note editor -->
```
This class is used to draw the icons visible under the text note editor

DevPhilB reviewed (changes requested):
Had to change the base branch to development 👎 
But in general, good job! 👍 
I've added some feedback (w/ solution) and as optional task you could change CSS class names, so that they are consistent too (it looks like you have already started it 😃 )

Commit:
Unify CSS class names and change comments

Simon-Deuring commented:
I implemented all your suggestions and changed CSS class names to be standardized.
But I still think that "to grant rights" is better than to "to give rights".

DevPhilB reviewed (approved):
👍
